src/main/protocol-handler/__test__/router.test.ts

```diff
@@ -33,49 +32,36 @@ function throwIfDefined(val: any): void {
 describe("protocol router tests", () => {
   let extensionInstances: ObservableMap<LensExtensionId, LensExtension>;
   let lpr: LensProtocolRouterMain;
-  let extensionsStore: ExtensionsStore;
+  let enabledExtensions: Set<string>;
 
   beforeEach(async () => {
     const di = getDiForUnitTesting({ doGeneralOverrides: true });
 
-    mockFs({
-      "tmp": {},
-    });
+    enabledExtensions = new Set();
 
-    di.override(extensionsStoreInjectable, () => ExtensionsStore.createInstance());
+    di.override(extensionsStoreInjectable, () => ({
+      isEnabled: ({ id, isBundled }: IsEnabledExtensionDescriptor) => isBundled || enabledExtensions.has(id),
+    } as unknown as ExtensionsStore));
 
     di.permitSideEffects(getConfigurationFileModelInjectable);
     di.permitSideEffects(appVersionInjectable);
 
     di.override(directoryForUserDataInjectable, () => "some-directory-for-user-data");
 
     extensionInstances = di.inject(extensionInstancesInjectable);
-    extensionsStore = di.inject(extensionsStoreInjectable);
     lpr = di.inject(lensProtocolRouterMainInjectable);
 
     lpr.rendererLoaded = true;
   });
 
-  afterEach(() => {
-    jest.clearAllMocks();
-
-    mockFs.restore();
-  });
-
   it("should throw on non-lens URLS", async () => {
-    try {
-      expect(await lpr.route("https://google.ca")).toBeUndefined();
-    } catch (error) {
-      expect(error).toBeInstanceOf(Error);
-    }
+    await lpr.route("https://google.ca");
```

Comment:
Are we changing the behaviour for a reason? If so, name of test lies. If not, the test is wrong.